src/collections/components/CollectionDetailsPage/form.tsx

```diff
@@ -74,12 +75,22 @@ function useCollectionUpdateForm(
     data: formData,
     triggerChange,
     setChanged,
-    hasChanged
-  } = useForm(getInitialData(collection, currentChannels));
+    hasChanged,
+    formId
+  } = useForm(getInitialData(collection, currentChannels), undefined, {
+    confirmLeave: true,
+    formId: COLLECTION_DETAILS_FORM_ID
+  });
 
-  const { setExitDialogSubmitRef, setEnableExitDialog } = useContext(
-    ExitFormDialogContext
-  );
+  const handleFormSubmit = useHandleFormSubmit({
+    formId,
+    onSubmit,
+    setChanged
+  });
+
+  const { setExitDialogSubmitRef } = useExitFormDialog({
+    formId: COLLECTION_DETAILS_FORM_ID
```

Comment:
Shouldn't it use formId returned by useForm?

Reply:
Since the forms that use `useChannels` have channels data completely outside of the form flow, this workaround is needed in order for exit prompt to work with such forms. Probably can be removed once we refactor the forms